src/components/MealPlanner.tsx

```
@@ -0,0 +1,120 @@
+    'midi': { recipeId: null, recipeName: null },
+    'soir': { recipeId: null, recipeName: null }
+};
+
+const MealPlanner: React.FC<MealPlannerProps> = ({ recipes }) => {
+    const [weeklyPlan, setWeeklyPlan] = useState<WeeklyPlan>(() => {
+        const savedPlan = localStorage.getItem('weeklyMealPlan');
+        if (savedPlan) {
+            return JSON.parse(savedPlan);
+        }
+        return DAYS_OF_WEEK.reduce((acc, day) => ({
+            ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)